### PR TITLE
VReplication Operator's guide first draft

### PR DESCRIPTION
**Pull request by sougou**
Signed-off-by: Sugu Sougoumarane <ssougou@gmail.com>

### CODE REVIEW

File: content/docs/advanced/vreplication.md [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+
+#### The pos field
+
+For starting a brand new vreplication stream, the `pos` field must be empty.
+The empty string signifies that there's no starting point for the vreplication.
+This causes VReplication to copy the contents of the source table first, and then
+start the replication.
+
+For large tables, this is done in chunks. After each chunk is copied, replication
+is resumed until it's caught up. Following this another chunk is copied, and so on,
```
Comment (deepthi) [inline content/docs/advanced/vreplication.md]: Worthwhile to note that only already copied rows are updated during replication phase.

File: content/docs/advanced/vreplication.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+until all tables are completed. After that, replication runs indefinitely.
+
+#### It's a shared row
+
+The vreplication table is a shared row. Once the row is creates, the VReplication stream
```
Comment (deepthi) [inline content/docs/advanced/vreplication.md]:
```suggestion
The vreplication table is a shared row. Once the row is created, the VReplication stream
```

File: content/docs/advanced/vreplication.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+
+The source vttablet sends its current time along with every event. This allows the
+target to correct for clock skew while estimating replication lag. Additionally,
+the source starts sending heartbeats if there is nothing to send. If the target
+receives no events from the source at all, it knows that it's definitely lagged
+and starts reporting itself accordingly.
+
+### Self-replication
+
+VReplication allows you the source keyspace/shard to be the same as the target.
```
Comment (deepthi) [inline content/docs/advanced/vreplication.md]:
```suggestion
VReplication allows you to set the source keyspace/shard to be the same as the target.
```

File: content/docs/advanced/vreplication.md [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+This is especially useful for schema rollouts. In this situation, and apply on
```
Comment (deepthi) [inline content/docs/advanced/vreplication.md]:
```suggestion
This is especially useful for schema rollouts. In this situation, an apply on
```

Comment (deepthi) [reply content/docs/advanced/vreplication.md]: How is it useful in schema rollouts?

File: content/docs/advanced/vreplication.md [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+It is possible that multiple streams can conflict with each other and cause
+deadlocks or lock waits. When such things happen, VReplication silently retries
+such transactions without reporting an error. It does increment a counter so
+that the frequency of such occurences can be tracked.
+
+### Automatic retries
+
+If any other error is encountered, the replication is retried after a short wait.
+Each time, the stream searches from the full list of available sources and picks
+one at randeom.
```
Comment (deepthi) [inline content/docs/advanced/vreplication.md]:
```suggestion
one at random.
```

**Review by deepthi — APPROVED** [review]
Very nicely written. A few typos/questions/suggestions inline.

File: content/docs/advanced/vreplication.md [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+  can resume the copy process
+  at a point that's consistent with the current replication position.
+* After copying is finished, it can continuously replicate the data
+  from the source to destination.
+* The copying rule can be expressed as a `select` statement. The
+  statement should be simple enough that the materialized table can
+  be kept up-to-date from the data coming from the binlog. For
+  example, joins are not supported.
+* Correctness verification (to be implemented): VReplication can
+  verify that the target table is an exact representaion of
```
Comment (gedgar) [inline content/docs/advanced/vreplication.md]:
```suggestion
  verify that the target table is an exact representation of
```

**Pull request merged**